sbi/utils/torchutils.py

```diff
@@ -208,7 +208,9 @@ def __init__(
 
 
 # XXX does an in-place version (e.g. make_conform_) make sense?
-def make_shapes_conform(parameter: torch.Tensor, observation: torch.Tensor) -> (torch.Tensor, torch.Tensor):
+def add_batch_dim(
```

Comment:
This function does not add the batch_dim unconditionally, but only under certain circumstances, that should be reflected in the name. But more importantly, now there's no dependency between the dimensions of parameters and observations, so both can be dealt with separately - in separate functions. I suggest ensure_parameter_batched and ensure_observation_batched but there are probably better names.